src/BigQuery/Job.php

```diff
@@ -121,7 +127,8 @@ public function cancel(array $options = [])
     }
```

Comment:
As far as I am aware there isn't really a de facto standard at the moment. There are abstractions users can tie into our library that implement event loops (https://github.com/reactphp/react is a great example) where they can wrap calls in promises, but our goal so far has been to do our best to provide base level compatibility with any execution environment.

We could do something like:

```php
$job = $bigQuery->runQueryAsJob($query);
$results = $job->queryResults();
$results->waitUntilComplete(); // blocks

foreach ($results->rows() as $row) {
    print_r($row);
}
```

But this requires an additional method call and I wasn't confident there was great value in having access to a `QueryResults` object if there were no rows available to access yet (since you can already access statistics/completion status on the `Job` object itself). What do you think?